### PR TITLE
Improve search overlay

### PR DESCRIPTION
**Pull request by olim88**
improve look of search overlay and added pet and star options.
# added background
image
image
# added max pet level and  otherwise add ```]``` to only get pets when searching for them
image
image
# added dungeon stars
image
image
# prioritise pets in auction house to show above skins
image

### CODE REVIEW

**Comment by UpFault** [comment]
I've been waiting to see this added, great pr imo

**Review by kevinthegreat1 — APPROVED** [review]
Great pr!

Even though it took a little bit of thinking to use the stars and pets feature. (I had to click the search suggestion.) Also, it seems like searching with stars easily exceeds the character count, and I don't seem to get what I searched for?

**Comment by Fluboxer** [comment]
while I didn't test it, I do have a suggestions:
1. Max out normal (and in 2nd action - master) stars if you hold shift 
2. Master stars should be shown as circle number, like they do ingame; use freed space to count stars 
изображение

File: src/main/java/de/hysky/skyblocker/skyblock/searchoverlay/OverlayScreen.java [hunk 1]
```diff
@@ -93,10 +119,85 @@ protected void init() {
...
+     */
+    @Override
+    public void renderBackground(DrawContext context, int mouseX, int mouseY, float delta) {
+        super.renderBackground(context, mouseX, mouseY, delta);
+        //find max height
+        int maxHeight = rowHeight * (1 + suggestionButtons.length + historyButtons.length);
+        if (historyButtons.length > 0) { //add space for history label if it could exist
+            maxHeight += (int) (rowHeight * 0.75);
+        }
+        context.drawGuiTexture(BACKGROUND_TEXTURE, searchField.getX() - 8, searchField.getY() - 8, (int) (this.width * 0.4) + 16, maxHeight + 16);
```
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/searchoverlay/OverlayScreen.java:198]: context.drawGuiTexture already does all the nine slice goodness. Case in point, the social interactions background's texture is only 236x34 but i don't see any stretching :p

**Comment by olim88** [comment]
> I had to click the search suggestion

it applies when the whole name of an item is what is search for (if you type the whole name it will still work). 
> searching with stars easily exceeds the character count, and I don't seem to get what I searched for

Yes when searching in the auction house the sign only has space for 30 characters. however the command (```/ahsearch```) gets around that and could be used instead however this would not work without cookie so i do not know another way to get past the limit
>yells in allignment noises

this is just because buttons have extra space for the highlight to go
image

>Max out normal (and in 2nd action - master) stars if you hold shift

I  do not fully understand this suggestion. hold shift while pressing enter? hold shift when clicking on the button? hold shift when selecting a dungeon item? just holding shift while in the menu?
>Master stars should be shown as circle number, like they do ingame; 

It is converted to that format when its outputted.  How would you be able to to input how many master stars you would like, when there is not a line of stars to click on, when using this method?

**Comment by Fluboxer** [comment]
> I do not fully understand this suggestion. hold shift while pressing enter? hold shift when clicking on the button? hold shift when selecting a dungeon item? just holding shift while in the menu?

>It is converted to that format when its outputted. How would you be able to to input how many master stars you would like, when there is not a line of stars to click on, when using this method?

oh, you can just hover and click? Well then forget what I said, I thought it is one of those silly designs where you need to click 10 times to get 10 stars selected

**Comment by kevinthegreat1** [comment]
> image _yells in allignment noises_

This alignment is correct, as olim pointed out.

> oh, you can just hover and click? Well then forget what I said, I thought it is one of those silly designs where you need to click 10 times to get 10 stars selected

No, you click the star you want. So I don’t think we need or can have your suggestions.

File: src/main/java/de/hysky/skyblocker/skyblock/searchoverlay/OverlayScreen.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/searchoverlay/OverlayScreen.java:198]: Oh, great.

**Comment by kevinthegreat1** [comment]
I just looked at the code and texture again. The social interactions background has one pixel of black border on the top and left but it has one pixel of white border on the bottom and right. So yes, it looks one pixel off, but I think it's fine as it.

**Pull request merged**